src/indexable.cr

```diff
@@ -699,7 +699,7 @@ module Indexable(T)
   end
 
   # Optimized version of `equals?` used when `other` is also an `Indexable`.
-  def equals?(other : Indexable, &) : Bool
+  def equals?(other : Indexable(U), & : (T, U) -> _) : Bool forall U
```

Comment:
The rest of the standard library doesn't seem to put parentheses around a `->` type name's parameters. I wonder if the formatter should remove them here